### PR TITLE
livecheck: split cask versions into sub-versions

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?
- [ ] Have you successfully run `brew man` locally and committed any changes?

-----

Follow-up to https://github.com/Homebrew/brew/pull/10348

Livecheck now correctly reports `15.0.1,9` as newer than `15,36`, but now `15.0.1,9.1` is reported as equal to `15.0.1,9`

This PR adds the `LivecheckVersion` class that compares the individual sub-versions of a formula/cask version, similar to how the `Version` class compares version tokens.

When comparing `15.0.1,9` and `15,36`:

1. `15.0.1` > `15`
1. Therefore, `15.0.1,9` > `15,36`

When comparing `15.0.1,9.1` and `15.0.1,9`:

1. `15.0.1` == `15.0.1`
1. `9.1` > `9`
1. Therefore, `15.0.1,9.1` > `15.0.1,9`

Related: https://github.com/Homebrew/homebrew-cask/pull/98130

### CODE REVIEW

File: Library/Homebrew/livecheck/livecheck.rb [hunk 1]
```diff
@@ -571,15 +571,63 @@ def latest_version(formula_or_cask, json: false, full_name: false, verbose: fals
...
-      when Formula
-        version
-      when Cask::Cask
-        Version.new(Cask::DSL::Version.new(version.to_s).before_comma)
-      else
-        T.absurd(formula_or_cask)
+    # A formula or cask version, split into its component sub-versions.
+    #
+    # @api private
+    class LivecheckVersion
```
Comment (MikeMcQuaid) [inline Library/Homebrew/livecheck/livecheck.rb:577]: Maybe a new `livecheck/version.rb` might be nicer for this?

File: Library/Homebrew/livecheck/version.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+      include Comparable
+
+      sig { params(formula_or_cask: T.any(Formula, Cask::Cask), version: Version).returns(LivecheckVersion) }
+      def self.create(formula_or_cask, version)
+        versions = case formula_or_cask
+        when Formula
+          [version]
+        when Cask::Cask
+          version.to_s.split(/[,:]/).map { |s| Version.new(s) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/livecheck/version.rb:20]: Not blocking but if there's a chance for tests for any of this that might be nice?

File: Library/Homebrew/livecheck/version.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    # A formula or cask version, split into its component sub-versions.
+    #
+    # @api private
+    class LivecheckVersion
```
Comment (MikeMcQuaid) [inline Library/Homebrew/livecheck/version.rb:9]:
```suggestion
    class Version
```

or rename the file `livecheck_version.rb` maybe?

File: Library/Homebrew/livecheck/version.rb (hunk 2 above)
Comment (vitorgalvao) [reply Library/Homebrew/livecheck/version.rb:20]: [We’re considering getting rid of `:` in `version`](https://github.com/Homebrew/homebrew-cask/issues/95207). Pointing that out so there’s something to link to from the other issue, and we can revisit this code when the time comes.

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**